pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
+1 for moving region/zone information to node.spec, instead of as a label, since this is the information which kubelet could get from cloud provider, just like `ExternalID` and `ProviderID`.

After all, if we make region/zone information a node label, why not make ExternalID and ProviderID as label, too?